src/ng-option-highlight/lib/ng-option-highlight.directive.ts

```diff
@@ -1,22 +1,23 @@
-import * as searchHelper from './search-helper';
-import { AfterViewInit, Directive, ElementRef, Input, OnChanges, Renderer2 } from '@angular/core';
+import { AfterViewInit, Directive, ElementRef, inject, input, OnChanges, Renderer2 } from '@angular/core';
 
 @Directive({
 	selector: '[ngOptionHighlight]',
 })
 export class NgOptionHighlightDirective implements OnChanges, AfterViewInit {
-	@Input('ngOptionHighlight') term: string;
-
+	term = input<string>(undefined, { alias: 'ngOptionHighlight' });
+	private elementRef = inject(ElementRef);
+	private renderer = inject(Renderer2);
 	private element: HTMLElement;
 	private label: string;
 
-	constructor(
-		private elementRef: ElementRef,
-		private renderer: Renderer2,
-	) {
+	constructor() {
 		this.element = this.elementRef.nativeElement;
 	}
 
+	private get _canHighlight() {
+		return this._isDefined(this.term()) && this._isDefined(this.label);
+	}
+
 	ngOnChanges() {
```

Comment:
@pavankjadda , since the "input" switched to Signal Input,  the hooks (ngOnInit, ngOnChanges) will never be triggered.

the alternative here is to use "effect" , so think the code under "ngOnChanges" & "ngAfterViewInit" should move under [effect](https://angular.dev/api/core/effect) or "[afterNextRender](https://angular.dev/api/core/afterNextRender)"